services/bettersocial/chat/updateBetterChannelMembers.js

```diff
@@ -72,21 +73,40 @@ const updateBetterChannelMembers = async (
   };
 };
 
+const __getAnonUserInfoPriority = (memberDataFromDb, member) => {
+  if (memberDataFromDb.anon_user_info_color_code) {
+    return {
+      color_name: memberDataFromDb.anon_user_info_color_name,
+      color_code: memberDataFromDb.anon_user_info_color_code,
+      emoji_name: memberDataFromDb.anon_user_info_emoji_name,
+      emoji_code: memberDataFromDb.anon_user_info_emoji_code
+    };
+  } else if (member.anon_user_info_color_code) {
+    return {
+      color_name: member.anon_user_info_color_name,
+      color_code: member.anon_user_info_color_code,
+      emoji_name: member.anon_user_info_emoji_name,
+      emoji_code: member.anon_user_info_emoji_code
+    };
+  } else {
+    const emoji = BetterSocialConstantListUtils.getRandomEmoji();
+    const color = BetterSocialConstantListUtils.getRandomColor();
+    return {
+      color_name: color.color,
+      color_code: color.code,
+      emoji_name: emoji.name,
+      emoji_code: emoji.emoji
+    };
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `__getAnonUserInfoPriority` has been added to determine the priority of anonymous user information based on certain conditions. It first checks if there's any existing anonymous user info from the database, then it checks if there's any anonymous user info from the member object, and finally, if neither exists, it generates random emoji and color for the anonymous user. This function seems to be well implemented and handles all possible scenarios.
```javascript
const __getAnonUserInfoPriority = (memberDataFromDb, member) => {
  if (memberDataFromDb.anon_user_info_color_code) {
    return {
      color_name: memberDataFromDb.anon_user_info_color_name,
      color_code: memberDataFromDb.anon_user_info_color_code,
      emoji_name: memberDataFromDb.anon_user_info_emoji_name,
      emoji_code: memberDataFromDb.anon_user_info_emoji_code
    };
  } else if (member.anon_user_info_color_code) {
    return {
      color_name: member.anon_user_info_color_name,
      color_code: member.anon_user_info_color_code,
      emoji_name: member.anon_user_info_emoji_name,
      emoji_code: member.anon_user_info_emoji_code
    };
  } else {
    const emoji = BetterSocialConstantListUtils.getRandomEmoji();
    const color = BetterSocialConstantListUtils.getRandomColor();
    return {
      color_name: color.color,
      color_code: color.code,
      emoji_name: emoji.name,
      emoji_code: emoji.emoji
    };
  }
};
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->